### PR TITLE
feat: use EvenEmitter3 for web-sdk

### LINKED ISSUE #845: [BUG] Removing skipLibCheck throws an error
### Observed behavior

We have a project where we want to use web-sdk and we already have a provider. When we try to use sdk in the project without the `--skipLibCheck` it doesn't compile.

It throws the following error:

```
Could not find a declaration file for module 'events'. 'relative-path/angular-first/node_modules/events/events.js' implicitly has an 'any' type.

Try `npm i --save-dev @types/events` if it exists or add a new declaration (.d.ts) file containing `declare module 'events';` [plugin angular-compiler]
```

By installing `npm i --save-dev @types/events`, I'm getting the following error:

```
TS2709: Cannot use namespace 'EventEmitter' as a type. [plugin angular-compiler]
node_modules/@openfeature/web-sdk/dist/types.d.ts:110:37:
protected readonly eventEmitter: EventEmitter;
```

### Expected Behavior

There is no compile error, even tho the skipLibCheck is false.

### Steps to reproduce

1. Create a new angular project - `ng new angular-test`
2. Install open-feature SDK - `npm i @openfeature/web-sdk`
3. Import Open Feature SDK in some TS file - `import { OpenFeature } from '@openfeature/web-sdk'`
4. Switch SkipLibCheck in tsconfig.json to **FALSE**

### PR DESCRIPTION
**Pull request by lukas-reining**
## This PR

Fixes an issue where the `events` node polyfill does not comply to the `node:events` types.
When trying to use the web OpenFeatureEventEmitter the following error message comes up, describing that the `events` polyfill's EventEmitter is incompatible to `node:events` EventEmitter.

```
✘ [ERROR] TS2416: Property 'eventEmitter' in type 'OpenFeatureEventEmitter' is not assignable to the same property in base type 'GenericEventEmitter<ProviderEmittableEvents, Record<string, unknown>>'.
  Type 'EventEmitter' is not assignable to type 'PlatformEventEmitter'.
    Types of property 'addListener' are incompatible.
      Type '(type: string | number, listener: Listener) => EventEmitter' is not assignable to type '(eventName: string | symbol, listener: (...args: any[]) => void) => PlatformEventEmitter'.
        Types of parameters 'type' and 'eventName' are incompatible.
          Type 'string | symbol' is not assignable to type 'string | number'.
            Type 'symbol' is not assignable to type 'string | number'.
```

This PR fixes that issue by not using the `events` anymore and instead using https://www.npmjs.com/package/eventemitter3

cc @toddbaert 

### Related Issues

Fixes #845

### CODE REVIEW

File: packages/shared/src/events/generic-event-emitter.ts [hunk 1]
```diff
@@ -94,13 +101,8 @@ interface PlatformEventEmitter {
...
   removeAllListeners(event?: string | symbol): this;
-  setMaxListeners(n: number): this;
-  getMaxListeners(): number;
   listeners(eventName: string | symbol): Function[];
-  rawListeners(eventName: string | symbol): Function[];
   emit(eventName: string | symbol, ...args: any[]): boolean;
   listenerCount(eventName: string | symbol, listener?: Function): number;
-  prependListener(eventName: string | symbol, listener: (...args: any[]) => void): this;
-  prependOnceListener(eventName: string | symbol, listener: (...args: any[]) => void): this;
   eventNames(): Array<string | symbol>;
```
Comment (lukas-reining) [inline packages/shared/src/events/generic-event-emitter.ts:107]: These had to be removed as EventEmitter3 removed these methods.
We do not need these and as `PlatformEventEmitter` is internal, it is not breaking to remove these methods.

https://www.npmjs.com/package/eventemitter3#eventemitter3:

> EventEmitter3 is a high performance EventEmitter. It has been micro-optimized for various of code paths making this, one of, if not the fastest EventEmitter available for Node.js and browsers. The module is API compatible with the EventEmitter that ships by default with Node.js but there are some slight differences:
>
>Domain support has been removed.
We do not throw an error when you emit an error event and nobody is listening.
The newListener and removeListener events have been removed as they are useful only in some uncommon use-cases.
The setMaxListeners, getMaxListeners, prependListener and prependOnceListener methods are not available.
Support for custom context for events so there is no need to use fn.bind.
The removeListener method removes all matching listeners, not only the first.

File: packages/shared/src/events/generic-event-emitter.ts [hunk 2]
```diff
@@ -33,7 +36,11 @@ export abstract class GenericEventEmitter<E extends AnyProviderEvent, Additional
     // The handlers have to be wrapped with an async function because if a synchronous functions throws an error,
     // the other handlers will not run.
     const asyncHandler = async (details?: EventDetails) => {
-      await handler(details);    
+      try {
+        await handler(details);
+      } catch (err) {
+        this._logger?.error('Error running event handler:', err);
+      }
```
[the remark this answers is not in the document]
Comment (lukas-reining) [reply packages/shared/src/events/generic-event-emitter.ts:43]: Functionally this is the same as before.

File: packages/shared/src/events/generic-event-emitter.ts (hunk 1 above)
Comment (toddbaert) [reply packages/shared/src/events/generic-event-emitter.ts:107]: Yep

File: packages/shared/src/events/generic-event-emitter.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (toddbaert) [reply packages/shared/src/events/generic-event-emitter.ts:43]: Yep again!

**Comment by toddbaert** [comment]
@lukas-reining @beeme1mr I pushed a small change to remove `@types/events` since it's not needed now.

**Review by toddbaert — APPROVED** [review]

**Comment by lukas-reining** [comment]
> @lukas-reining @beeme1mr I pushed a small change to remove `@types/events` since it's not needed now.

Ah sure missed that one thanks!

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**